compiler/lib/eval.ml

```diff
@@ -237,7 +237,7 @@ let the_cont_of info x (a : cont array) =
 let constant_js_equal a b =
   match a, b with
   | Int i, Int j -> Some (Int32.equal i j)
-  | Float a, Float b -> Some (Poly.equal a b)
+  | Float a, Float b -> Some (Float.equal a b)
```

Comment:
In JavaScript, `NaN == NaN` is false, but `Float.equal nan nan` is true.

Reply:
fixed. Now using `Float.ieee_equal` alias to `Poly.equal`